ruma-api-macros/src/api/response.rs

```diff
@@ -131,10 +131,17 @@ impl Response {
                         }
                     }
                     _ => quote! {
-                        resp_builder = resp_builder.header(
-                            #import_path::exports::http::header::#header_name,
-                            response.#field_name,
-                        );
+                        resp_builder.headers_mut()
```

Comment:
This returns `None` "when builder has error". I don't think that can be triggered by our generated code, so I think we should use `.unwrap()` or `.expect(...)` rather than `.map`.

Reply:
Also I think we should obtain the mutable headermap reference once outside `apply_header_fields` and only access it here.